safehttp/defaults/defaults.go

```diff
@@ -0,0 +1,49 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// 	https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Pacakge defaults provides ready to use, safe, pre-configured instances of safehttp types.
+package defaults
+
+import (
+	"github.com/google/go-safeweb/safehttp"
+	"github.com/google/go-safeweb/safehttp/plugins/coop"
+	"github.com/google/go-safeweb/safehttp/plugins/cors"
+	"github.com/google/go-safeweb/safehttp/plugins/csp"
+	"github.com/google/go-safeweb/safehttp/plugins/fetchmetadata"
+	"github.com/google/go-safeweb/safehttp/plugins/hostcheck"
+	"github.com/google/go-safeweb/safehttp/plugins/hsts"
+	"github.com/google/go-safeweb/safehttp/plugins/staticheaders"
+	"github.com/google/go-safeweb/safehttp/plugins/xsrf/xsrfhtml"
+)
+
+// ServeMuxConfig creates a safe and ready to use ServeMuxConfig with all necessary security interceptors installed.
+func ServeMuxConfig(xsrfKey string, hosts, allowedCORSOrigins []string) *safehttp.ServeMuxConfig {
```

Comment:
Maybe this could benefit from go/gotip/episodes/41?

Reply:
I thought about this but solution 2 doesn't really fit, and if you adopt solution 1 and you add one more parameter every user of this function will keep working without realizing the API has changed. With that solution there is no way to make parameters mandatory.

Reply:
Yup, you're right.

In this case, it seems to be more natural to me to have them in the `hosts, xsrf, cors` order. Alternatively, just `hosts, xsrf` and simply not support CORS in the default mux. WDYT?

Reply:
Agreed!